tools/c7n_azure/c7n_azure/function_package.py

```diff
@@ -197,38 +203,68 @@ def publish(self, deployment_creds):
 
         self.log.info("Function publish result: %s" % r.status_code)
 
-    def wait_for_remote_build(self, deployment_creds, is_consumption):
-        self.log.info('Waiting for the remote build to finish')
-
-        # Replicate the behavior from azure core func tool.. Some racing in kudulight,
-        # so different way to get the status. https://bit.ly/32AM71b
-        if not is_consumption:
-            is_deploying = True
-            is_deploying_uri = '%s/api/isdeploying' % deployment_creds.scm_uri
-            while is_deploying != 'False':
-                self.log.info('Waiting for deployment to complete...')
-                is_deploying = requests.get(is_deploying_uri).json()['value']
-                time.sleep(10)
-
-        # Get deployment id
+    def wait_for_remote_build(self, deployment_creds):
+        self.log.info("Appending deployment info")
+        if not FunctionPackage.disable_wait_for_build and not deployment_creds_at_exit:
+            atexit.register(self.wait_for_remote_builds, deployment_creds_at_exit)
```

Comment:
why is this an at exit (process exit), thats makes a lot of assumptions on the caller, where this is a library, generally I'd expect to see at an exit, much closer to the process entry point/controller.

Reply:
See my reply.. I see it now as a cheap way to add `Provider.dispose`, but I'm 100% willing to do the right way (like let providers do their cleanup..)